### ISSUE TITLE
Limiting spectator camera modes

### ISSUE THREAD

**Issue by thepaullee** [CombinedArmsGaming/F3_CA#93]
Thoughts on limiting spectator camera modes for unconscious and/ or dead spectators? Limited to only 1PP and 3PP so you can't free roam around the map 

Now that we have unconscious spectate to help with the downtime, it unfortunately does create some awkward moments where you see where all the enemies are, who's dead, etc and then pop back in the action once you're awake (or respawned in the case of being dead). 

This is pretty immersion breaking IMO and could be handled nicely by limiting spectate to follow-only

**Comment by dgibso29** [comment]
@Bubbus and I have talked about it a bit. What it comes down to is that free cam is only an issue when they're unconscious. If they're dead, by the time they respawn and make it back to the fight, anything they saw is generally now out of date. The ideal solution would be to have different limitations on dead vs unconscious spectators, though I do not know how feasible that is in practice.

Another possible way to resolve this would be limiting unconscious freecam to a relatively small area around the downed person's body, but again I don't know how feasible that is in reality.

If none of the above are possible, then I think that limiting spectate to 1PP/3PP is the correct option. I am considering doing exactly that for all spectators for my mission on the 6th, but haven't decided yet.

**Comment by thepaullee** [comment]
I agree with Daniel

**Comment by Poulern** [comment]
The only feasible solution as I see it for a "quick fix" would be to make it limited to 1PP/3PP. Still doesn't address the whole able to talk to other dead people or being able to see things in TvTs for example.

**Comment by jimshacklock** [comment]
To play devils advocate somewhat:

If you start limiting unconscious people's ability to see/talk/do stuff whilst unconscious does that not somewhat defeat the point of the whole unconscious spectator script in the first place (allowing people to have something to do whilst down).

As an aside, a semi-related nice to have feature would making it an optional thing to go into the spectator mode. I've heard a few people being a bit surprised/confused by it, and I can see the argument that it does take you out of the immersion rather a lot. 3PP/Limited radius spectate around your body could be a sweet spot?

**Comment by thepaullee** [comment]
I think limiting uncon spectator to 1PP/3PP so you can only see what other players see is a good first step, half the time I think people just chill talking to other players that are in spectator as well

**Comment by Poulern** [comment]
There is also a worry about spectator chat turning into a salt chat too. 

As for limiting it you can change this number to a really high one (or maybe even put it to 60s?) and it will only go into spectator when you die.
![image](https://user-images.githubusercontent.com/7356386/106671363-ab71aa00-65ae-11eb-9885-ea9f86b795fe.png)

**Comment by jimshacklock** [comment]
Maybe that's a good starting point - set that to like 2 mins or something so people who go down for prolonged periods can get the benefit but it's not catching every single person

**Comment by Bubbus** [comment]
Regarding some folks not liking the cam, I didn't see that coming, sorry about that.  I heard some folks express this on Sunday and I'm playing with this right now - hope it helps out (will be testing on my framework branch fairly soon):

![Addon option](https://i.imgur.com/mkyPvSo.png)

For reference, 16 seconds is the value I found to be the "minimum sure duration" that you're going to stay down for this version of medical.  Setting it higher than that just means people are spending longer staring at nothing which is what this feature was designed to eliminate.  As we've seen in the last few ops this is subject to taste, some people might like to stay down and out for longer so I'm looking into adding a client option for that too.

RE: 1PP/3PP and avoiding salty spectator, I have seen people abusing freecam more often that I'm personally happy with but I haven't experienced any greater levels of frustration in the spectator chat than was normal before.
As a developer I know it's tempting to force user behaviour via code but at the end of the day we're a mature community and while people should hopefully never feel constrained, I think we do expect some decorum via social norms.  
I feel conflicted about taking toys away from everyone because of the few.  Specifically, if people are degrading the experience in general we have more direct means of tackling that.

I recognise I'm biased, but I want to ask that we carefully consider against restricting this feature too heavily.  The initial design intent was a bit selfish: to stop me reading news articles for a quarter of my game time because they were simply more interesting than my game.  Since this feature got merged into the main framework, I can easily say I've been having more fun.

**Comment by jimshacklock** [comment]
Yeah I definitely agree that it's a net gain having it in the base framework. A Clientside setting is probably the ideal way to go here really

**Comment by thepaullee** [comment]
For dead-dead spectate I think keeping freecam is fine, but a bit of a limited view while in unconscious spectate would go a long way 

It might be worth probing some players about but from what I understand the dislike/ confusion for the unconscious cam comes from situations where you can see the enemies who just downed you, only to come back awake a few moments later and having omniscient knowledge on everything around you

**Comment by yozer247** [comment]
Keep in mind forced first-person and third-person means you're constantly listening to the alive players talk, so if someone wants to find a spot for silence they can't.
